api/v1alpha1/contour_types.go

```diff
@@ -302,6 +317,75 @@ const (
 	GCPLoadBalancerProvider   LoadBalancerProviderType = "GCP"
 )
 
+type AwsLoadBalancerParameters struct {
+	// AllocationID is a list of Allocation IDs of Elastic IP addresses that are
+	// to be assigned to created Network Load Balancer.
+	// Example: "eipalloc-<xxxxxxxxxxxxxxxxx>"
+	//
+	// See: https://docs.aws.amazon.com/eks/latest/userguide/load-balancing.html
+	//
+	// +optional
+	AllocationID []string `json:"allocationId,omitempty"`
+}
+
+type AzureLoadBalancerParameters struct {
+	// Address is the desired IP of LoadBalancer. If LoadBalancerScope is Internal,
+	// IP address must reside in same virtual network as AKS and must not already
+	// be assigned to a resource. If IP address does not reside in same subnet as
+	// AKS, subnet parameter is also required.
+	//
+	// See:
+	// 	 https://docs.microsoft.com/en-us/azure/aks/static-ip#create-a-service-using-the-static-ip-address
+	// 	 https://docs.microsoft.com/en-us/azure/aks/internal-lb#specify-an-ip-address
+	//
+	// xref: https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha1/gateway_types.go#L492-L498
```

Comment:
Sorry for the confusion, we don't want to xref in the docs. I would like to use similar kubebuilder validation tags for this field:
```
	// +kubebuilder:validation:MinLength=1
	// +kubebuilder:validation:MaxLength=253
```

Reply:
Fixed